### PR TITLE
OpenShift image + support

### PR DESCRIPTION
**Pull request by pohly**
The Dockerfile.UBI file will be pulled by the Red Hat build service. We just have to include it in our repo. Alternatively, the image can be built locally and pushed to the Red Hat partner registry.

In the driver itself we need changes to support the scheduler extensions on OpenShift 4.6 and 4.7.

Fixes: #652

### CODE REVIEW

**Comment by pohly** [comment]
@avalluri: can you perhaps review?

I may have to take out the ipmctl commit before merging, depending on when we get the necessary paperwork done.

**Comment by pohly** [comment]
> I may have to take out the ipmctl commit before merging, depending on when we get the necessary paperwork done.

I pulled out that one commit and moved it into https://github.com/intel/pmem-csi/pull/1026/

File: test/e2e/deploy/deploy.go
Comment (pohly) [inline test/e2e/deploy/deploy.go:1309]: This breaks skew testing because older releases didn't have this container port entry.

I'm currently testing a fix.

@avalluri: do you want me to force-push right away or wait until you are done with your review?

**Review by avalluri — APPROVED** [review]

File: hack/set-version.sh
Comment (pohly) [reply hack/set-version.sh:27]: I need to fix that comment. The plan no longer is to include the generated files on release branches. Updating the operator bundle always seems to be a manual process. One can enable the autobuilder, but its input must be prepared manually and then one might as well do everything manually.

Comment (pohly) [reply hack/set-version.sh:27]: With that in mind it's probably better to keep the current state and not commit the generated files.

File: pkg/ndctl/region.go
Comment (pohly) [reply pkg/ndctl/region.go:206]: 16MiB.

Comment (pohly) [reply pkg/ndctl/region.go:206]: I'm wondering whether we should be even more conservative and fall back to 96MiB. That's what was needed for https://github.com/intel/pmem-csi/issues/964

Comment (pohly) [reply pkg/ndctl/region.go:206]: I've switched to 96MiB and also refactored the code so that both namespace creation and capacity retrieval use the same helper function. We had similar (but not quite the same!) code in both.

Comment (avalluri) [reply pkg/ndctl/region.go:206]: After going through issue #964, I feel like using 'no. of interleaves * 16MiB' as alignment is a good solution than fixing the alignment to 96MiB.

File: hack/set-version.sh
Comment (avalluri) [reply hack/set-version.sh:27]: Okay.

File: pkg/ndctl/region.go
Comment (pohly) [reply pkg/ndctl/region.go:206]: But using that as alignment was exactly the problem in #964 - we did that and ignored the region alignment, which then didn't work.

The lesson from #964 is that we have to consider *two* alignment constraints. The problem here is that the kernel is not reporting the second constraint (region alignment), so we have to make some educated guess.

**Comment by pohly** [comment]
> avalluri approved these changes 

I've pushed an update. @avalluri: please take another look at 0211dfc, the updated commit for the region alignment fallback.

I rebased on top of https://github.com/intel/pmem-csi/pull/1028, therefore the overall diff (https://github.com/intel/pmem-csi/compare/076196ab4b15db467ca2340b1b5573e849902d6f..1070c89f97e36c48a739a9e3a6c3f6094639f280) is a bit larger.

**Pull request merged**
